### DISCUSSION TITLE
An option for `get_table_names` to return fully qualified table names

### DISCUSSION THREAD

**Discussion by no23reason** [duckdb/duckdb#11019] [Ideas]
When using the Python `DuckDBPyConnection.get_table_names` function with queries that query across different schemas, the output is not enough to identify the particular tables.

For example for this (admittedly a bit strange yet valid) query:
```sql
SELECT c.name as customer_name, r.name as representative_name
FROM customer.individuals c FULL JOIN representative.individuals r
ON p.representative_id = r.id
```
The following python code
```py
duckdb.default_connection.get_table_names(query)
```
returns a set containing just one item: `individuals` even though there are two different tables at play.

I believe it could be useful to add some option to make the `get_table_names` return the names in a fully-qualified form, including the schema. For example:
```py
duckdb.default_connection.get_table_names(query, include_schema=True)
```
would return set with `customer.individuals` and `representative.individuals`.

**Comment by taniabogatsch** [comment]
I'll likely pick this up soon! I will ping here if I do.

**Comment by taniabogatsch** [comment]
https://github.com/duckdb/duckdb/pull/17472

PR for the C++ side / C API is up here - in a follow-up, we can propagate these changes to the Python client.

**Reply by aszenz** [reply]
Would this be available in wasm as well? 

On a side note are there any plans to have get_column_names as well? 
I would like too know the tables and their columns for a sql query

**Reply by taniabogatsch** [reply]
I haven't planned to PR the changes to WASM, but happy to review a PR (should be a minor change).

As for `get_column_names`, that might be possible with the `TableRef` (`tableref.hpp`), which has:
```cpp
	//! Aliases for the column names
	vector<string> column_name_alias;
```
Again, happy to review a PR. :) :D

**Comment by taniabogatsch** [comment]
Closing this as resolved in https://github.com/duckdb/duckdb/pull/17472 and https://github.com/duckdb/duckdb/pull/17797
